docs(manifest): Improve `theme_color` member page (#35643)
mdn/content: files/en-us/web/manifest/theme_color/index.md

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

> The `theme_color` member specifies the default theme color for your web application's user interface.
> This color may be applied to various browser UI elements, such as the toolbar, address bar, and status bar.

Reviewer: You may already have a pre-agreed pattern for this whole page. FWIW I much prefer this to what has gone before.

That said, I really like the pattern of having a summary line/paragraph where possible. So for me this would be better:

```suggestion
The `theme_color` member specifies the default theme color for your web application's user interface.

This color may be applied to various browser UI elements, such as the toolbar, address bar, and status bar.
```

Author: I see what you mean. I'd prefer to have the two sentences about where the effect could be noticed right after the first one. I've moved the last line about "native app-like experience" to a new para.

Reviewer: That makes sense.

> This color may be applied to various browser UI elements, such as the toolbar, address bar, and status bar.

Reviewer: The definition feels a little circular - "a theme color is a theme color". Consider in the description having some images? Just a comment.

Author: I know and I agree about the circular reference. I'd considered this quite a bit, switching between "primary color" and "primary theme color". I finally decided on using "default theme color" because the [spec](https://w3c.github.io/manifest/#theme_color-member) seems to make a special mention of it, with a [definition](https://w3c.github.io/manifest/#dfn-default-theme-color) linking back to `theme_color` :). I wasn't sure if there is any special significance of the term so I decided to lean towards the spec.

But thinking about it again now. Perhaps just saying "default color for your web app's UI" might do the trick! WDYT?

> Consider in the description having some images?

Indeed, it's on my mind. It's something I want to do for the `display` manifest member as well, to show a side-by-side comparison. I'll probably tackle it in a separate round. (there's one on https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name/theme-color#example, but I was hoping to could create something new)

Reviewer: Works for me! Thanks.

It can be particularly noticeable in contexts like the task switcher or when the app is added to the home screen.
This color application can provide a more native app-like experience for your web app, especially when it's loaded in [standalone](/en-US/docs/Web/Manifest/display#standalone) mode.

## Syntax

```json
"theme_color": "colorValue"
```

### Values

- `colorValue`

>   - : A string that specifies a [valid color value](/en-US/docs/Web/CSS/color_value).

Reviewer: I don't know why, but this is not rendering as a definition list.

Author: Sorry about that. Fixed in https://github.com/mdn/content/pull/35643/commits/18aa935f8590cf83c0fe38c1df4b2e1ce065e536

  > [!NOTE]
  > Browsers may ignore the alpha component of the color based on the context.
  > In most environments, `theme_color` cannot be transparent.
  > It's recommended to use fully opaque colors (alpha value of 1 or 100%) to ensure consistent behavior across different platforms and browsers.

## Description

While optional, specifying a `theme_color` allows you to extend your app's visual identity beyond its content areas.

> Choose a `theme_color` that aligns with your app's brand guidelines, which can enhance user recognition and recall, particularly when your app is viewed alongside other applications or system interfaces.

Reviewer: ```suggestion
Choose a `theme_color` that aligns with your app's brand guidelines, as this can enhance user recognition and recall, particularly when your app is viewed alongside other applications or system interfaces.
```

In browsers that support `theme_color`, the value specified in the manifest file serves as the default theme color for your web app across all pages where the manifest is applied.
You can override this default in the following ways:

> - Using the [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value of the `name` attribute in the HTML `<meta>` element: You can specify a theme color for a web page that's different from the manifest `theme_color` specified for your app. This enables you to set unique theme colors for individual pages within your app.

Reviewer: ```suggestion
- Using the [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value of the `name` attribute in the HTML `<meta>` element: You can specify a theme color for a web page that's different from the manifest `theme_color` specified for your app. This enables you to set different theme colors for individual pages within your app.
```

> ```html
> <meta name="theme-color" content="#9370DB" />
> ```

Reviewer: Alignment
```suggestion
  ```html
  <meta name="theme-color" content="#9370DB" />
  ```
```

Author: Yes! Done

- Combining the `<meta name="theme-color">` element with media queries: You can specify the theme color to be used based on user's color scheme preference.

```html
<meta
  name="theme-color"
  content="#F4E6D8"
  media="(prefers-color-scheme: light)" />
<meta
  name="theme-color"
  content="#5D4037"
  media="(prefers-color-scheme: dark)" />

> ```

Reviewer: Also should align.

Author: Done

These override methods provide you the flexibility to adapt your app's appearance for specific pages or user preferences, improving the overall user experience.

Browsers may also adjust the applied theme color based on user preferences.

> If a user has set a preference for light or dark mode, browsers may override the manifest `theme_color` value to support any [`prefers-color-scheme`](/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query defined in your app's CSS.

Reviewer: Perhaps 

```suggestion
If a user has set a preference for light or dark mode, browsers may override the manifest `theme_color` value to better match any [`prefers-color-scheme`](/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query defined in your app's CSS.
```

Author: In this case, I think it might be better to stick to what spec has, which is "support"

Reviewer: Support is definitely "sub optimal", but perhaps "comply with" would be better. Not worth arguing about.

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

The `theme_color` member specifies the default color for your web application's user interface.
This color may be applied to various browser UI elements, such as the toolbar, address bar, and status bar.
It can be particularly noticeable in contexts like the task switcher or when the app is added to the home screen.

This color application can provide a more native app-like experience for your web app, especially when it's loaded in [standalone](/en-US/docs/Web/Manifest/display#standalone) mode.

## Syntax

```json

> "theme_color": "colorValue"

Author: Thanks for this suggestion, @hamishwillee!
See my whole spiel here https://github.com/mdn/content/pull/35557#issuecomment-2313416707 about how to name the values and why I initially didn't go for `< >`. But with an additional vote from you, I am going to switch the usage to angle brackets.
I'm also going to use a hyphen in the placeholder values.

```suggestion
"theme_color": "<color-value>"
```

```

### Values

> - `colorValue`

Reviewer: In other cases the list is the property name - here you have it as the description

```suggestion
- `theme_color`
```

  ```

- Combining the `<meta name="theme-color">` element with media queries: You can specify the theme color to be used based on user's color scheme preference.

  ```html
  <meta
    name="theme-color"
    content="#F4E6D8"
    media="(prefers-color-scheme: light)" />
  <meta
    name="theme-color"
    content="#5D4037"
    media="(prefers-color-scheme: dark)" />
  ```

These override methods provide you the flexibility to adapt your app's appearance for specific pages or user preferences, improving the overall user experience.

Browsers may also adjust the applied theme color based on user preferences.
If a user has set a preference for light or dark mode, browsers may override the manifest `theme_color` value to support any [`prefers-color-scheme`](/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query defined in your app's CSS.

```css
body {
  background: #f8f9fa;
  color: #212529;
}

@media (prefers-color-scheme: dark) {
  body {
    background: #212529;
    color: #f8f9fa;
  }
}
```

## Examples

### Using a named color

```json
"theme_color": "red"
```

### Using an RGB value

```json
"theme_color": "rgb(66, 133, 244)"
```

### Using a hexadecimal value

```json
{

>   "name": "MyWebApp",

Author: this shouldn't be in camel case
```suggestion
  "name": "My First App",
```
